bank_statement_line_attachment/models/account_bank_statement.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2019 Eficent Business and IT Consulting Services S.L.
+#        (http://www.eficent.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class AccountBankStatementLine(models.Model):
+    _inherit = 'account.bank.statement.line'
+
+    attachment_number = fields.Integer('Attachments',
+                                       compute='_compute_attachment_number')
+    requires_attachment = fields.Boolean('Requires attachment',
+                                         default=False)
+
+    @api.multi
+    def _compute_attachment_number(self):
+        attachment_data = self.env['ir.attachment'].read_group(
+            [('res_model', '=', 'account.bank.statement.line'),
+             ('res_id', 'in', self.ids)], ['res_id'], ['res_id'])
```

Comment:
```suggestion
             ('res_id', 'in', self.ids)], ['res_id'], ['res_id:sum'])
```
it will count on sql level so you don't need to itereate again on python level